helm/chart/router/templates/deployment.yaml

```
@@ -129,6 +129,9 @@ spec:
         {{- if .Values.extraContainers }}
           {{- include "common.tplvalues.render" (dict "value" .Values.extraContainers "context" $) | nindent 8 }}
         {{- end }}
+        {{- if .Values.initContainers }}
```

We need to update our deployment template to indicate that these containers are `initContainers`, otherwise they will just be processed as ordinary containers with bad results (such as causing the pod to fail when the init container terminates).

I'll modify the PR to fix this.

```suggestion
        {{- if .Values.initContainers }}
        initContainers:
```